site/app/libraries/FileUtils.php

```diff
@@ -480,6 +480,43 @@ public static function areWordsInFile(string $file, array $words) {
         return $words_detected;
     }
 
+    public static function checkZipFileStatus($file){
+        $zip = new \ZipArchive();
+        //open file with additional checks
+        $res = $zip->open($file, \ZipArchive::CHECKCONS);
```

Comment:
let's just return here with the status code, and see below note about splitting the printing of error messages into ErrorMessages